### ISSUE TITLE
RFE: Instead of username/password using certificates - possible?

### ISSUE THREAD

**Issue by oaxjogi** [hschott/ready-websocket-plugin#8]
HI,

would it be possible to add certificates for opening a websocket connection? I would also spend some of my time to help to get the things done.

Yours,
Jogi

**Comment by alcalda** [comment]
Upvote from me. Need also possibility to connect to secure websocket servers using a client certificate.

**Comment by hschott** [comment]
Should already work. Did you try to set KeyStore and KeyStore Password within SoapUI Preferences.

**Comment by oaxjogi** [comment]
Yes, it works by adding correct certificates in the JRE which SOAP UI using.

**Comment by hschott** [comment]
OK, this works of course. But my question was, does it also work when adding a Keyfile within the SoapUI Preferences, which is the standard approach for mutual authentication with SoapUI.

**Comment by sashat720** [comment]
Suppose you want to open a websocket to wss://echo.websocket.org, which certificate should I install in the JRE which SoapUI is using into which keystore?
(maybe in jre\lib\security\cacerts ?)

**Comment by rarsov** [comment]
Hi , I am trying to open a new Websocket Connection using wss, and i keep receiving the following:
Cancelling due to failed test step:java.util.concurrent.ExecutionException: javax.websocket.DeploymentException: Handshake error.], time taken = 302
I ve tried to export the certificate that I need and use it as a keystore in the SSL settings it did not work.
I added the certificate to the jre\lib\security\cacerts file and used that as a keystore, it recognizes it and the status is ok, but I keep getting the same handshake error when I try to open a new connection

**Comment by hschott** [comment]
Did you try the documented way?https://support.smartbear.com/readyapi/docs/projects/requests/ssl.html
